repos/system_upgrade/el7toel8/actors/addupgradebootentry/libraries/library.py

```diff
@@ -10,6 +10,14 @@
 def add_boot_entry():
     debug = 'debug' if os.getenv('LEAPP_DEBUG', '0') == '1' else ''
 
+    args = debug + ' enforcing=0 rd.plymouth=0 plymouth.enable=0'
+
+    with open('/proc/cmdline') as f:
+        fips = 'fips=1' in f.read()
+
+    if fips:
```

Comment:
```suggestion
    fips = next(self.consume(FIPSFacts))
    if fips.enabled:
```
Something like this could be enough and you could remove the _propagatefips_ actor.

Reply:
Actually the actor will be probably needed. This is not something what should be in the addupgradebootentry actor. The actor should be consuming special msg, based on which will do the thing. As it is highly possible we will need more stuff in future like that for cmdline - as well, users would needed to be able to extend the IPU solution. But ok to keept it there for now, as solution is not created yet.